engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -430,17 +434,45 @@ case class WorkflowExecutionActor(params: WorkflowExecutionActorParams)
     )
   }
 
+   /*
+    * If this ExpressionKey has an AnonymousExpressionNode that feeds a task call input, use the backend IoFunctionSet
+    * instead of the engine's IoFunctionSet to properly handle `writeFile` or `readFile` invocations.
+    */
+  private def processRunnableTaskCallInputExpression(key: ExpressionKey,
+                                                     data: WorkflowExecutionActorData,
+                                                     anonymousExpressionNode: AnonymousExpressionNode): ErrorOr[WorkflowExecutionDiff] = {
+    import cats.syntax.either._
+
+    val callNode = anonymousExpressionNode.callNodeReceivingInput.get(())
+
+    (for {
+      taskCallNode <- (Option(callNode) collect { case t: TaskCallNode => t }).toChecked(s"Not a TaskCallNode: $callNode")
+      jobKey <- data.executionStore.keyForNode(callNode).toChecked(s"No job key found for call node $callNode")
+      backendJobDescriptorKey <- Option(jobKey).collectFirst({ case c: BackendJobDescriptorKey => c }).toChecked(s"Job key not a BackendJobDescriptorKey: $jobKey")
+      factory <- backendFactoryForTaskCallNode(taskCallNode)
+      backendInitializationData = params.initializationData.get(factory.name)
+      functions = factory.expressionLanguageFunctions(workflowDescriptor.backendDescriptor, backendJobDescriptorKey, backendInitializationData)
+      diff <- key.processRunnable(functions, data.valueStore, self).toEither
+    } yield diff).toValidated
+  }
+
+  private def backendFactoryForTaskCallNode(taskCallNode: TaskCallNode): Checked[BackendLifecycleActorFactory] = {
+    for {
+      name <- workflowDescriptor
+        .backendAssignments.get(taskCallNode).toChecked(s"Cannot find an assigned backend for call ${taskCallNode.fullyQualifiedName}")
+      factory <- backendFactories.get(name).toChecked(s"Cannot find a backend factory for backend $name")
+    } yield factory
```

Comment:
TOL curious if there’s a reason for splitting this out from the main for comp?

Reply:
This chunk of logic is called in two spots, the evaluate-an-anonymous-expression-that-feeds-a-task-call code and the actual okay-now-actually-go-run-the-task-call code.